src/silly_sprites/opengl/gl_window.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#define GLFW_INCLUDE_NONE
+#include <GLFW/glfw3.h>
+#include <spdlog/spdlog.h>
+#include <tl/expected.hpp>
+#include "gl_error.hpp"
+
+namespace sly::gl {
+	
+	class Window final {
+	private:
+		GLFWwindow* m_window;
+
+    public:
+		Window(GLFWwindow* window);
```

Comment:
The constructor should be private if you want to force the user to use the `create` function.

Reply:
> The constructor should be private if you want to force the user to use the `create` function.

i'm not able to make this ctor private and delete the other ctors.
i dont see a way to return the window in `crate()` then. when an other 'ctor' is default the 'dtor' will destroy the window.

is there any workaround?